# Load the data

In [1]:
import numpy as np
import pandas as pd

In [6]:
dataset_backup = pd.read_excel("statistika.xlsx", sheet_name="Sheet1")

In [7]:
dataset = dataset_backup.copy()
dataset.head()

RB  SPOL  MJESTO   DOB  RAZLOG/dg  OTPUST  2 DANA  30 DANA  ICU  SMRT  ...  \
0   1     1       1  1936          6       1     1.0      1.0    0     0  ...   
1   2     2       1  1972         10       1     3.0      4.0    0     0  ...   
2   3     2       1  1946          9       4     5.0      1.0    1     0  ...   
3   4     2       1  1927         11       6     1.0      2.0    0     0  ...   
4   5     2       1  1934         10       3     3.0      1.0    0     0  ...   

   KISIK_BOD_6  TLAK_6  TLAK_BOD_6  PULS_6  PULS_BOD_6  SVIJEST_6  \
0          NaN     NaN         NaN     NaN         NaN        NaN   
1          NaN     NaN         NaN     NaN         NaN        NaN   
2          NaN     NaN         NaN     NaN         NaN        NaN   
3          NaN     NaN         NaN     NaN         NaN        NaN   
4          NaN     NaN         NaN     NaN         NaN        NaN   

   SVIJEST_BOD_6  TEMPERATURA_6  TEMP_BOD_6  UKUPNI_BODOVI_6  
0            NaN            NaN         NaN              NaN  
1            NaN            NaN         NaN              NaN  
2            NaN            NaN         NaN              NaN  
3            NaN            NaN         NaN              NaN  
4            NaN            NaN         NaN              NaN  

[5 rows x 114 columns]

In [11]:
dataset.shape

(278, 114)

In [12]:
dataset.columns.values

array(['RB', 'SPOL', 'MJESTO', 'DOB', 'RAZLOG/dg', 'OTPUST', '2 DANA',
       '30 DANA', 'ICU', 'SMRT', 'RIZIK_POJEDINACNI', 'RIZIK_UKUPNI',
       'RESPIRACIJE_1', 'RESP_BOD_1', 'SPO2_1', 'SPO2_BOD_1', 'SPO22_1',
       'SPO22_BOD_1', 'KISIK_1', 'KISIK_BOD_1', 'TLAK_1', 'TLAK_BOD_1',
       'PULS_1', 'PULS_BOD_1', 'SVIJEST_1', 'SVIJEST_BOD_1',
       'TEMPERATURA_1', 'TEMP_BOD_1', 'UKUPNI_BODOVI_1', 'RESPIRACIJE_2',
       'RESP_BOD_2', 'SPO2_2', 'SPO2_BOD_2', 'SPO22_2', 'SPO22_BOD_2',
       'KISIK_2', 'KISIK_BOD_2', 'TLAK_2', 'TLAK_BOD_2', 'PULS_2',
       'PULS_BOD_2', 'SVIJEST_2', 'SVIJEST_BOD_2', 'TEMPERATURA_2',
       'TEMP_BOD_2', 'UKUPNI_BODOVI_2', 'RESPIRACIJE_3', 'RESP_BOD_3',
       'SPO2_3', 'SPO2_BOD_3', 'SPO22_3', 'SPO22_BOD_3', 'KISIK_3',
       'KISIK_BOD_3', 'TLAK_3', 'TLAK_BOD_3', 'PULS_3', 'PULS_BOD_3',
       'SVIJEST_3', 'SVIJEST_BOD_3', 'TEMPERATURA_3', 'TEMP_BOD_3',
       'UKUPNI_BODOVI_3', 'RESPIRACIJE_4', 'RESP_BOD_4', 'SPO2_4',
       'SPO2_BOD_4', 'SPO22_4

In [13]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 278 entries, 0 to 277
Columns: 114 entries, RB to UKUPNI_BODOVI_6
dtypes: float64(81), int64(33)
memory usage: 247.7 KB


# Analysis

## Sex of patients

In [15]:
dataset["SPOL"].unique()

array([1, 2])

In [59]:
dataset["spol_mz"] = dataset["SPOL"].apply(str).replace("1", "M")
dataset["spol_mz"] = dataset["spol_mz"].replace("2", "Z")

In [60]:
dataset["spol_mz"]

0      M
1      Z
2      Z
3      Z
4      Z
      ..
273    M
274    Z
275    Z
276    Z
277    M
Name: spol_mz, Length: 278, dtype: object

In [61]:
freq_table = pd.crosstab(dataset['spol_mz'], 'frequency')
freq_table["rel_frequency"] = freq_table['frequency']/len(dataset["spol_mz"])
freq_table

col_0    frequency  rel_frequency
spol_mz                          
M              126       0.453237
Z              152       0.546763

In [71]:
freq_table["frequency"]["M"]

126

In [65]:
from scipy.stats import binom_test

In [72]:
binom_test(x=freq_table["frequency"]["M"], n=len(dataset["spol_mz"]), p=1/2, alternative="two-sided")

/var/folders/pn/n2g0nqvs69b0yx721gj7v9gw0000gn/T/ipykernel_1559/2888726873.py:1: DeprecationWarning: 'binom_test' is deprecated in favour of 'binomtest' from version 1.7.0 and will be removed in Scipy 1.12.0.
  binom_test(x=freq_table["frequency"]["M"], n=len(dataset["spol_mz"]), p=1/2, alternative="two-sided")


0.1336244676178877

Na razini znacajnosti alpha=0.05 testirajmo hipotezu da uzorak dolazi iz populacije s jednakim brojem zena i muskaraca. Ako s p oznacimo vjerojatnost da je slucajno odabrana osoba muskarac, onda testiramo nulhipotezu H_0 i njoj suprotstavljamo alternativnu hipotezu H_1, gdje su one dane s:\
H_0: p=1/2 \
H_1: p!=1/2

Dobivena p-vrijednosti iznosi 0.1336244676178877 > alpha, sto znaci da na razini znacajnosti od 0.05 nemamo razloga odbaciti nulhipotezu. Stoga na razini znacajnosti 0.05 nemamo razloga sumnjati da uzorak dolazi iz binomne distribucije s parametrom 1/2. 